File: src/dataset/javascript/grade-school_exercism-javascript-bearguns/gemini/GeminiFlash_grade-school_v4.js
'use strict'
class School {
  constructor() {
    this.fullroster = {};
  }

  roster() {
    return Object.assign({}, this.fullroster);
  }

  add(name, grade) {
    if (!this.fullroster[grade]) {
      this.fullroster[grade] = [];
    }
    this.fullroster[grade].push(name);
    this.fullroster[grade].sort();
  }

  grade(grade) {
    const roster = this.fullroster[grade];
    if (!roster) {
      return [];
    }
    return [...roster].sort();
  }
}

export default School;

File: src/dataset/cpp/grade-school_exercism-johnngugi/gemini/GeminiFlash_grade_school_v4.cpp
#include "grade_school.h"
#include <algorithm>
#include <vector>
#include <map>
#include <string>

using namespace std;
using namespace grade_school;

map<int, vector<string>> school::roster() const
{
    return grades;
}

void school::add(const string& name, int grade_level)
{
    auto& grade_list = grades[grade_level];
    grade_list.push_back(name);
    std::sort(grade_list.begin(), grade_list.end());
}

vector<string> school::grade(int grade_level) const
{
    auto it = grades.find(grade_level);
    if (it != grades.end()) {
        return it->second;
    } else {
        return {};
    }
}